iac-common/src/testFixtures/java/org/sonar/iac/common/testing/Verifier.java

```diff
@@ -244,22 +250,44 @@ public void reportIssue(HasTextRange toHighlight, String message, List<Secondary
     }
 
     protected void reportIssue(TextRange textRange, String message, List<SecondaryLocation> secondaryLocations) {
-      Issue issue = new Issue(textRange, message, secondaryLocations);
+      var issue = new Issue(textRange, message, secondaryLocations);
       if (!raisedIssues.contains(issue)) {
         TextPointer start = textRange.start();
         TextPointer end = textRange.end();
-        SingleFileVerifier.Issue reportedIssue = verifier
+
+        var reportedIssue = ((SingleFileVerifier) verifier)
           .reportIssue(message)
           .onRange(start.line(), start.lineOffset() + 1, end.line(), end.lineOffset());
-        secondaryLocations.forEach(secondary -> reportedIssue.addSecondary(
-          secondary.textRange.start().line(),
-          secondary.textRange.start().lineOffset() + 1,
-          secondary.textRange.end().line(),
-          secondary.textRange.end().lineOffset(),
-          secondary.message));
+
+        secondaryLocations.forEach(secondary -> {
+          if (secondary.filePath != null) {
+            addSecondaryOnDifferentFile(((MultiFileVerifier.Issue) reportedIssue), secondary);
+          } else {
+            addSecondaryOnMainFile(reportedIssue, secondary);
+          }
+        });
         raisedIssues.add(issue);
       }
     }
+
+    private static void addSecondaryOnMainFile(SingleFileVerifier.Issue reportedIssue, SecondaryLocation secondary) {
+      reportedIssue.addSecondary(
+        secondary.textRange.start().line(),
+        secondary.textRange.start().lineOffset() + 1,
+        secondary.textRange.end().line(),
+        secondary.textRange.end().lineOffset(),
+        secondary.message);
+    }
+
+    private static void addSecondaryOnDifferentFile(MultiFileVerifier.Issue reportedIssue, SecondaryLocation secondary) {
+      reportedIssue.addSecondary(
+        Path.of(secondary.filePath),
+        secondary.textRange.start().line(),
+        secondary.textRange.start().lineOffset() + 1,
+        secondary.textRange.end().line(),
+        secondary.textRange.end().lineOffset(),
+        secondary.message);
+    }
```

Comment:
This approach is necessary / chossen because of the way the `InternalIssueVerifier` API is implemented.
Basically `InternalIssueVerifier` implements `MultiFileVerifier` and `SingleFileVerifie`r, so it supports both.
It uses `InternalIssue` that implements `MultiFileVerifier.Issue` and `SingleFileVerifier.Issue`, but we can't use InternalIssue because it's not public...

Reply:
In theory, we could still be using the `SingleFileVerifier` instead of the `InternalIssueVerifier`, as it's basically the same.